rewrite-java/src/main/java/org/openrewrite/java/format/NormalizeLineBreaksVisitor.java

```diff
@@ -79,7 +79,7 @@ private static String normalizeNewLines(String text, boolean useCrlf) {
                 normalized.append(c);
             }
         }
-        return normalized.toString();
+        return normalized.toString().equals(text) ? text : normalized.toString();
```

Comment:
Note: There may be a similar bug to what happened with whitespace.
`normalized.toString()` may be identical to `text`, but will create an extra cyle without this check.